rest/routes/identity.go

```diff
@@ -0,0 +1,28 @@
+package routes 
+
+import (
+  "encoding/json"
+  "net/http"
+
+  "github.com/RedHatInsights/chrome-service-backend/rest/util"
+  "github.com/RedHatInsights/chrome-service-backend/rest/service"
+  "github.com/RedHatInsights/chrome-service-backend/rest/models"
+  "github.com/go-chi/chi/v5"
+)
+
+func GetUserIdentity(w http.ResponseWriter, r *http.Request) {
+  user := r.Context().Value(util.USER_CTX_KEY).(models.UserIdentity) 
+  userID := user.ID
+  response := make(map[string]models.UserIdentity)
+  
+  userData, err := service.GetUserIdentityByID(userID)
```

Comment:
The context under `util.USER_CTX_KEY` actually gives you the user identity already. It does a `FirstOrCreate` lookup, that will either retrieve the entity or creates it. The logic is in the `InjectUser` middleware.

Essentially you are doing the same query twice.